### PR TITLE
Move request provider code to a separate package + Raise PHP to 8.1

### PR DESCRIPTION
**Pull request by arogachev**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -31,8 +31,9 @@
             "url": "https://github.com/sponsors/yiisoft"
         }
     ],
+    "minimum-stability": "dev",
```
Comment (samdark) [inline composer.json:34]: Remove before release.

Comment (arogachev) [reply composer.json:34]: Most likely it will be removed in a separate PR, so I created according issue - #29.

File: composer.json [hunk 2]
```diff
@@ -41,7 +42,8 @@
         "yiisoft/hydrator": "^1.0",
         "yiisoft/hydrator-validator": "^1.0",
         "yiisoft/middleware-dispatcher": "^5.1",
-        "yiisoft/validator": "^1.1"
+        "yiisoft/validator": "^1.1",
+        "yiisoft/request-provider": "dev-master"
```
Comment (vjik) [inline composer.json:46]:
```suggestion
        "yiisoft/request-provider": "^1.0"
```

File: composer.json (hunk 1 above)
Comment (vjik) [inline composer.json:34]:
```suggestion
```

**Review by vjik — APPROVED** [review]

**Pull request merged**